src/System.Runtime/tests/System/Buffer.cs

```diff
@@ -4,111 +4,211 @@
+            new object[] { typeof(uint), sizeof(uint) },
+            new object[] { typeof(long), sizeof(long) },
+            new object[] { typeof(ulong), sizeof(ulong) },
+            new object[] { typeof(IntPtr), sizeof(IntPtr) },
+            new object[] { typeof(UIntPtr), sizeof(UIntPtr) },
+            new object[] { typeof(double), sizeof(double) },
+            new object[] { typeof(float), sizeof(float) },
+            new object[] { typeof(bool), sizeof(bool) },
+            new object[] { typeof(char), sizeof(char) }
+        };
     }
 
-    [Theory, MemberData(nameof(ByteLengthTestData))]
+    [Theory]
+    [MemberData(nameof(ByteLength_TestData))]
     public static void TestByteLength(Type type, int size)
     {
-        const int length = 25;
-        Array array = Array.CreateInstance(type, length);
-        if (type.GetTypeInfo().IsPrimitive)
+        const int Length = 25;
+        Array array = Array.CreateInstance(type, Length);
+        Assert.Equal(Length * size, Buffer.ByteLength(array));
+    }
+
+    [Fact]
+    public static void TestByteLength_Invalid()
+    {
+        Assert.Throws<ArgumentNullException>("array", () => Buffer.ByteLength(null)); // Array is null
+
+        Assert.Throws<ArgumentException>("array", () => Buffer.ByteLength(Array.CreateInstance(typeof(DateTime), 25))); // Array is not a primitive
+        Assert.Throws<ArgumentException>("array", () => Buffer.ByteLength(Array.CreateInstance(typeof(decimal), 25))); // Array is not a primitive
+        Assert.Throws<ArgumentException>("array", () => Buffer.ByteLength(Array.CreateInstance(typeof(string), 25))); // Array is not a primitive
+    }
+
+    [Theory]
+    [InlineData(new uint[] { 0x01234567, 0x89abcdef }, 0, 0x67)]
+    [InlineData(new uint[] { 0x01234567, 0x89abcdef }, 7, 0x89)]
+    public static void TestGetByte(Array array, int index, int expected)
+    {
+        Assert.Equal(expected, Buffer.GetByte(array, index));
+    }
+
+    [Fact]
+    public static void TestGetByte_Invalid()
+    {
+        var array = new uint[] { 0x01234567, 0x89abcdef };
+
+        Assert.Throws<ArgumentNullException>("array", () => Buffer.GetByte(null, 0)); // Array is null
+        Assert.Throws<ArgumentException>("array", () => Buffer.GetByte(new object[10], 0)); // Array is not a primitive array
+
+        Assert.Throws<ArgumentOutOfRangeException>("index", () => Buffer.GetByte(array, -1)); // Index < 0
+        Assert.Throws<ArgumentOutOfRangeException>("index", () => Buffer.GetByte(array, 8)); // Index >= array.Length
+    }
+
+    [Theory]
+    [InlineData(0, 0)]
+    [InlineData(0, 5000)]
+    [InlineData(5000, 6000)]
+    public static unsafe void TestMemoryCopy_Long(int sourceIndexOffset, int destinationIndexOffset)
+    {
+        var sourceArray = new int[25000];
+        for (int i = 0; i < sourceArray.Length; i++)
+        {
+            sourceArray[i] = i;
+        }
+
+        var destinationArray = new int[30000];
```

Comment:
Could you put the 25000 and the 30000 into named consts, that are then used here and below?